src/org/intellij/erlang/inspection/ErlangUndefinedCallbackFunctionInspection.java

```diff
@@ -64,26 +65,30 @@ protected void checkFile(@NotNull ErlangFile file, @NotNull ProblemsHolder probl
         builder.append("'").append(ErlangPsiImplUtil.createFunctionPresentationFromCallbackSpec(spec)).append("', ");
       }
       String message = builder.substring(0, builder.length() - 2);
-      registerProblem(problemsHolder, behaviourRef, message, new MyLocalQuickFixBase(undefinedCallbacks));
+      registerProblem(problemsHolder, behaviourRef, message, new CreateAndExportFunctionsFix(undefinedCallbacks));
     }
   }
 
-  private static class MyLocalQuickFixBase extends LocalQuickFixBase {
+  private static class CreateAndExportFunctionsFix extends LocalQuickFixBase {
     @NotNull
-    private final Collection<ErlangCallbackSpec> myCallbackSpecs;
+    private final Collection<SmartPsiElementPointer<ErlangCallbackSpec>> myCallbackSpecs;
 
-    protected MyLocalQuickFixBase(@NotNull Collection<ErlangCallbackSpec> callbackSpecs) {
+    public CreateAndExportFunctionsFix(@NotNull Collection<ErlangCallbackSpec> callbackSpecs) {
       super(FIX_MESSAGE);
-      myCallbackSpecs = callbackSpecs;
+      myCallbackSpecs = ContainerUtil.map(callbackSpecs, new Function<ErlangCallbackSpec, SmartPsiElementPointer<ErlangCallbackSpec>>() {
+        @Override
+        public SmartPsiElementPointer<ErlangCallbackSpec> fun(ErlangCallbackSpec callbackSpec) {
+          return SmartPointerManager.getInstance(callbackSpec.getProject()).createSmartPsiElementPointer(callbackSpec);
```

Comment:
You only need to obtain SmartPointerManager instance once. Why doing it for each callback you are to implement?